### PR TITLE
Arg makefile dialyzer tweaks

### PR DESCRIPTION
**Pull request by argv0**
added a standard rebar-wrapper makefile w/useful targets for testing outside of a riak tree
exported some types to stop dialyzer from complaining

### CODE REVIEW

**Comment by argv0** [comment]
dunno why the commits are appearing twice.   i'll send a better PR later :(

File: src/riak_pipe_vnode.erl [hunk 1]
```diff
@@ -741,6 +743,7 @@ restart_worker(Worker, State) ->
             [ reply_to_blocker(Blocker, fail)
               || {_, Blocker} <- queue:to_list(Worker#worker.blocking) ],
             %% ask fitting to find a better place for the other inputs
+            %% XXX ARG: riak_pipe_fitting:send_fail/2 doesnt seem to exist?
```
Comment (beerriot) [inline src/riak_pipe_vnode.erl]: That's correct -- send_fail/2 is TODO work.  If you look at riak_pipe_vnode_worker:send_output/3,4, you'll find that handing the 'fail' messages sent by line 741/743 above is TODO work as well.  This is a large part of https://agilezen.com/project/17206/story/312

**Comment by beerriot** [comment]
I missed this PR due to misconfigured notifications on this repo, so the work done in src/*.erl was duplicated by https://github.com/basho/riak_pipe/pull/1

The rebar.config, .gitignore, Makefile, and dialyzer ignores look good to me, though.  (When do we finally get to leave the rebar script out of these repos?)  Rebase or deal with merge conflicts?

**Comment by beerriot** [comment]
Is https://github.com/basho/riak_pipe/pull/4 an acceptable cherry-picking?

**Comment by beerriot** [comment]
https://github.com/basho/riak_pipe/pull/4 accepted as substitute

**Pull request closed without merging**